tests/e2e/lifecycle/pvc_clone/test_pgsql_pvc_clone.py

```diff
@@ -0,0 +1,113 @@
+        """
+        1. Deploy PGSQL workload
+        2. Create multiple clone of same PVC when the PVC usage is different
+        3. Attach a new pgsql pod to it.
+        4. Create pgbench benchmark to new pgsql pod
+        """
+
+        # Deploy PGSQL workload
+        log.info("Deploying pgsql workloads")
+        pgsql = pgsql_factory_fixture(replicas=3, clients=3, transactions=600)
+
+        # Get postgres pvcs obj list
+        postgres_pvcs_obj = pgsql.get_postgres_pvc()
+
+        # Get postgres pods obj list
+        postgres_pods_obj = pgsql.get_postgres_pods()
+
+        self.sset_list = []
+        for i in range(3):
+
+            # Get usage of pgsql pvc
+            parent_pods_obj = pgsql.get_postgres_used_file_space(postgres_pods_obj)
+
+            # Create clone of pgsql pvc
+            log.info("Creating clone of the Postgres PVCs")
+            cloned_pvcs = [pvc_clone_factory(
+                pvc_obj, volume_mode=VOLUME_MODE_FILESYSTEM
+            ) for pvc_obj in postgres_pvcs_obj]
+            log.info("Created clone of the PVCs and all cloned PVCs are in Bound state")
+
+            # Attach to new postgres pod
+            self.pgsql_obj_list = pgsql.attach_pgsql_pod_to_claim_pvc(
+                pvc_objs=cloned_pvcs, postgres_name=f'postgres-cloned-{i}',
+                run_benchmark=False
+            )
+            self.sset_list.extend(self.pgsql_obj_list)
+
+            # Validate cloned pvcs file space matches with parent
+            cloned_pods_list = get_pod_name_by_pattern(
+                pattern=f'postgres-cloned-{i}', namespace=RIPSAW_NAMESPACE
+            )
+            cloned_pods_obj = [
+                get_pod_obj(name=pods, namespace=RIPSAW_NAMESPACE) for pods in cloned_pods_list
+            ]
+            cloned_obj = pgsql.get_postgres_used_file_space(cloned_pods_obj)
+            for pod_obj in parent_pods_obj:
+                if pod_obj.filespace != cloned_obj[parent_pods_obj.index(pod_obj)].filespace:
+                    raise Exception(
+                        f"Parent pvc {pod_obj.name} used file space is {pod_obj.filespace}. "
+                        f"And for cloned pvc {cloned_obj[parent_pods_obj.index(pod_obj)].name} "
+                        f"used file space is {cloned_obj[parent_pods_obj.index(pod_obj)].filespace}"
+                    )
+            log.info("All cloned PVC matches the parent PVC data")
+
+            # Run benchmark on cloned PVC
+            pgsql.create_pgbench_benchmark(
+                replicas=3, pgbench_name=f'pgbench-cloned-{i}',
+                postgres_name=f'postgres-cloned-{i}', wait=False
+            )
+
+            # Run benchmark on parent PVC
+            pgsql.create_pgbench_benchmark(replicas=3, pgbench_name=f'pgbench-{i}', wait=False)
+
+            # Wait till pgbench client pods up
+            wait_time = 180
+            log.info(f"Waiting {wait_time} seconds for pgbench client pods to be up")
+            time.sleep(180)
```

Comment:
Can we use wait_for_pgbench_status?

Reply:
The problem here is pgbench client pod takes some time to come up. So if we use `wait_for_pgbench_status` immediately it checks for existing pgbench-client pods only.